File: examples/key/encrypt_rsa/encrypt_rsa.py
from bloock.entity.encrypter import RsaEncrypter
from bloock.client.encryption import EncryptionClient
from bloock.client.record import RecordClient

payload = "This will be encrypted"
record_client = RecordClient()
encryption_client = EncryptionClient()
key_client = KeyClient()

print(f'The following payload will be encrypted: "{payload}"')

key = key_client.new_local_key(KeyType.Rsa2048)

# To encrypt a record during the building process
encrypted_record = (
    record_client.from_string(payload)
    .with_encrypter(RsaEncrypter(EncrypterArgs(key)))
    .build()
)

# To encrypt a record independently
record = record_client.from_string(payload).build()
encrypted_record = encryption_client.encrypt(record, RsaEncrypter(EncrypterArgs(key)))

print("Encryption successful")
print(f"Encrypted payload: {encrypted_record.retrieve().decode()}")

File: examples/key/encrypt_rsa/encrypt_rsa.java
import com.bloock.sdk.client.EncryptionClient;
import com.bloock.sdk.client.RecordClient;
import com.bloock.sdk.entity.Record;
import com.bloock.sdk.entity.RsaEncrypter;
import com.bloock.sdk.entity.RsaKeyPair;

import java.nio.charset.StandardCharsets;

public class App {
  public static void main(String[] args) throws Exception {
    String payload = "This will be encrypted";

    System.out.println("The following payload will be encrypted: " + payload);
    RecordClient recordClient = new RecordClient();
    KeyClient keyClient = new KeyClient();

    LocalKey localKey = keyClient.newLocalKey(KeyType.Rsa2048);

    // To encrypt a record during the building process
    Record encryptedRecord = recordClient.fromString(payload)
        .withEncrypter(new RsaEncrypter(localKey))
        .build();

    // To encrypt a record independently
    EncryptionClient encryptionClient = new EncryptionClient();
    Record record = recordClient.fromString(payload).build();
    encryptedRecord = encryptionClient.encrypt(record, new RsaEncrypter(localKey));

    System.out.println("Encryption successful");
    System.out.println("Encrypted payload: " + new String(record.retrieve(), StandardCharsets.UTF_8));
  }
}